lua/gitlab/actions/pipeline.lua

```diff
@@ -73,6 +78,59 @@ M.retrigger    = function()
   end)
 end
 
+M.see_logs     = function()
+  local bufnr = vim.api.nvim_get_current_buf()
+  local linnr = vim.api.nvim_win_get_cursor(0)[1]
+  local text = u.get_line_content(bufnr, linnr)
+  local last_word = u.get_last_chunk(text)
+  if last_word == nil then
+    vim.notify("Cannot find job name", vim.log.levels.ERROR)
+    return
+  end
+
+  local j = nil
+  for _, pipeline_job in ipairs(M.pipeline_jobs) do
+    if pipeline_job.name == last_word then j = pipeline_job end
+  end
+
+  if j == nil then
+    vim.notify("Cannot find job in state", vim.log.levels.ERROR)
+    return
+  end
+
+  local body = { job_id = j.id }
+  job.run_job("/job", "GET", body, function(data)
+    local file = data.file
+    if file == "" then
+      vim.notify("Log trace is empty", vim.log.levels.WARN)
+      return
+    end
+
+    local lines = {}
+    for line in file:gmatch("[^\n]+") do
+      table.insert(lines, line)
+    end
+
+    if #lines == 0 then
+      vim.notify("Log trace lines could not be parsed", vim.log.levels.ERROR)
+      return
+    end
+
+    M.pipeline_popup:unmount()
+    vim.cmd.enew()
+
+    bufnr = vim.api.nvim_get_current_buf()
+    vim.api.nvim_buf_set_lines(bufnr, 0, -1, false, lines)
+
+    local job_file_path = string.format("/tmp/gitlab.nvim.job-%d", j.id)
+    vim.cmd("w! " .. job_file_path)
```

Comment:
This use of a temporary file and use of `term cat` let us use the ANSII escape codes to color/style the output of the logfile easily without having to parse out individual ANSII escape codes and so forth.

However, the use of the `/tmp` folder will probably be another breaking point of compatibility with Windows and should get picked up in https://github.com/harrisoncramer/gitlab.nvim/pull/35